### DISCUSSION TITLE
Last 1 - 3 LEDs flashing intermediately

### DISCUSSION THREAD

**Discussion by Schnielz** [awawa-dev/HyperHDR#495] [Q&A] [answered]
Hello All,

I have a relative common HyperHDR setup

RPi: Raspberry Pi 2 Model B V1.1 RAM 1GB
APA102 LEDs (130 LEDs)
HyperHDR 19 beta 2
Dedicated 5V power supply for LEDs connected to the strip on 2 positions. Voltage on the strip never lower than 4.85V.

Everything works fine but the last 1 - 3 LEDs are flashing white unregulary every couple of seconds. The last one more often the others less. When I disable the LEDs in the config the flashing stops. I recently replaced my WS2801 LEDs which had a similar problem but these didn't stop flashing even when disabled in the config.

I don't have a level shifter installed. LED ground is connected to the PI as recommended.

Has anyone an idea what the reason for the flashing could be?

Here is the config:
[HyperHDR-19.0.0.0beta2-Backup-2023-01-25 APA102.json.txt](https://github.com/awawa-dev/HyperHDR/files/10508059/HyperHDR-19.0.0.0beta2-Backup-2023-01-25.APA102.json.txt)

**Comment by awawa-dev** [comment]
Hi
Quite unusual problem. Could you try to increase SPI speed to 4000000 or 12000000 and in your HyperHDR LED strip configuration increase from 130 LEDs to 134-150 or so and play some effects to test if it changes anything? Power injection at the end of the LED strip is usually recommended.

**Comment by Schnielz** [comment, marked as answer]
Unfortunately both didn't help. But now I recognized that there is flashing a small white bar in the bottom right corner popping up in the live view of the LED visualization. Most likely it comes from the Ezcap 269 grabber then. It also only happens when dark colors are shown. In the USB capture screen I cannot configure very much. Most stuff is "Automatic" as I can't find the 269 in the drop down.

Now I worked around it by just shorten the capture region of these LEDs a bit.

![214948270-8b7ba312-62f3-4c9d-b4c1-9a5792385b09](https://user-images.githubusercontent.com/14457580/215080668-306636fc-6182-4f5d-ba15-b81b81e1e967.png)

EDIT: Replaced the image to show the white bar at the correct position.

**Comment by awawa-dev** [comment]
Does it happen when you have HDR tone mapping enabled? On Linux Ezcap 269 allows rather full control: in the grabber tab for resolutions & codecs use the yellow button next to the device in the select list (you should not use automatic device detection but select ezcap 269 from the list, avoid MJPEG it's ugly on Ezcap269). Then you can control also brightness/contrast/saturation and hue in fields below.
![obraz](https://user-images.githubusercontent.com/69086569/214953341-d02543a4-e6f4-4189-bfde-b34170855b61.png)

**Comment by awawa-dev** [comment]
you can also use bottom cropping there to remove last lines.

**Comment by Schnielz** [comment]
It happens regardless of HDR tone mapping enabled or not. Unfortunately I cannot find Ezcap 269 in the list:
![image](https://user-images.githubusercontent.com/14457580/215166169-7916e65c-755a-4ba2-945a-4c07399d3d66.png)

**Comment by awawa-dev** [comment]
It should be 'Ezcap U3 ... (video0)' on your screenshot. In your current HyperHDR logs you should find information that it opens /dev/video0 device while initializing the USB grabber.

**Comment by awawa-dev** [comment]
I forget that on USB2.0 Ezcap 269 doesn't support YUV encoding unfortunately..only MJPEG. YUV needs USB3.0

**Comment by Schnielz** [comment]
Thank you very much for your help! I am quite happy with the current result. Maybe I switch over to a RPi3 or 4 one day for USB3 but I am actually satisfied and RPi prices are three times higher than 2 years back. Have a nice weekend!

**Comment by awawa-dev** [comment]
Yes, Rpi prices are crazy, availability is low and if nothing improves in the near future, this platform is in danger of going out of business. An alternative may be ARM Amlogic Tv boxes with USB 3.0, there are also new, very interesting grabbers on the horizon, but it is always a difficult decision whether to update our systems, because we can always encounter new problems. Especially if we're happy with it. Have a nice weekend too.

**Reply by reedy-au** [reply]
second hand intel NUC? ebay has them with SSD 4GB ram for $40-50usd. works great 👍
